File: questPresentation/src/main/java/ru/nekit/android/qls/quest/types/VisualRepresentationalNumberSummandQuest.java
package ru.nekit.android.qls.quest.types;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;

import ru.nekit.android.qls.quest.resources.common.IVisualRepresentationHolder;

public class VisualRepresentationalNumberSummandQuest extends NumberSummandQuest
        implements IVisualRepresentationHolder {

    @Nullable
    List<Integer> mVisualRepresentationList;

    public VisualRepresentationalNumberSummandQuest() {
        mVisualRepresentationList = new ArrayList<>();
    }

    @Nullable
    @Override
    public List<Integer> getVisualRepresentationList() {
        return mVisualRepresentationList;
    }

    @Override
    public void setVisualRepresentationList(@NonNull List<Integer> value) {
        mVisualRepresentationList = value;
    }
}

File: questDomain/src/main/java/ru/nekit/android/qls/domain/model/quest/VisualRepresentationalNumberSummandQuest.kt
package ru.nekit.android.qls.domain.model.quest

import ru.nekit.android.qls.domain.model.resources.common.IVisualRepresentationHolder

open class VisualRepresentationalNumberSummandQuest : NumberSummandQuest(), IVisualRepresentationHolder {

    override var visualRepresentationList: MutableList<Int> = ArrayList()

}